CompoundCalc/Models/Requests/InterestCalcReq.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Newtonsoft.Json;
+
+namespace CompoundCalc.Models.Requests;
+
+public class InterestCalcReq
+{
+    public InterestCalcReq(double startingBalance,
+     double interestRate, int years)
+    {
+        StartingBalance = startingBalance;
+        InterestRate = interestRate;
+        Years = years;
+
+        ValidateParams();
+    }
+
+    [Required(ErrorMessage = "Starting balance is required")]
+    [JsonProperty("startingBalance")]
```

Comment:
i dont think you need jsonproperty here since they are the same as the prop name